src/elm/quantity.coffee

```diff
@@ -250,22 +252,26 @@ module.exports.parseQuantity = (str) ->
 
 module.exports.doAddition = (a,b) ->
   if a instanceof Quantity and b instanceof Quantity
+    [a_unit, b_unit] = [coalesceToOne(a.unit), coalesceToOne(b.unit)]
     # The units don't have to match (m and m^2), but must be convertable
     # we will choose the unit of a to be the unit we return
-    val = convert_value(b.value, b.unit, a.unit)
+    val = convert_value(b.value, b_unit, a_unit)
     return null unless val?
-    new Quantity({unit: a.unit, value: a.value + val})
+    new Quantity({unit: a_unit, value: a.value + val})
   else
-    a.copy?().add?(b.value, clean_unit(b.unit))
+    b_unit = if b instanceof Quantity then coalesceToOne(b.unit) else b.unit
+    a.copy?().add?(b.value, clean_unit(b_unit))
```

Comment:
The only classes I can find with an `add` property are `DateTime` and `Date`.  Is that what this is intended to handle?  If so, should we be checking for it (via `else if` instead of just the `else`)?